app/controllers/api/order_cycles_controller.rb

```diff
@@ -15,26 +17,36 @@ def products
 
       render json: products
     rescue ProductsRenderer::NoProducts
-      render status: :not_found, json: ''
+      render_no_products
     end
 
     def taxons
-      taxons = Spree::Taxon.
-        joins(:products).
-        where(spree_products: { id: distributed_products }).
-        select('DISTINCT spree_taxons.*')
+      taxons = Rails.cache.fetch("oc-taxons-#{order_cycle.id}", expires_in: 30.seconds) do
+        Spree::Taxon.
+          joins(:products).
+          where(spree_products: { id: distributed_products }).
+          select('DISTINCT spree_taxons.*')
+      end
 
       render json: ActiveModel::ArraySerializer.new(taxons, each_serializer: Api::TaxonSerializer)
     end
 
     def properties
+      properties = Rails.cache.fetch("oc-properties-#{order_cycle.id}", expires_in: 30.seconds) do
```

Comment:
can we use a constant instead of a magic number for the `expires_in`? It will make future changes slightly easier.

Reply:
I realised we can use ActionCaching here instead (avoid the magic number issue in a similar way). It's much cleaner, and I think it's more efficient as it caches the data _after_ the output from the render call has been generated.

Reply:
Hmmm, turns out fancy Action Caching doesn't play nicely with API controllers in Rails. It's got issues with any controller that inherits from `ActionController::Metal` instead of `ActionController::Base` and the workarounds don't look very nice.

Reply:
Update: we can use Action Caching in the API.